x/lockup/keeper/keeper.go

```diff
@@ -0,0 +1,60 @@
+package keeper
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	paramstypes "github.com/cosmos/cosmos-sdk/x/params/types"
+
+	"github.com/althea-net/althea-chain/x/lockup/types"
+)
+
+type Keeper struct {
+	storeKey   sdk.StoreKey
+	paramSpace paramstypes.Subspace
+	cdc        codec.BinaryMarshaler
+}
+
+func NewKeeper(cdc codec.BinaryMarshaler, storeKey sdk.StoreKey, paramSpace paramstypes.Subspace) Keeper {
+	// set KeyTable if it has not already been set
+	if !paramSpace.HasKeyTable() {
```

Comment:
Do I need this subspace? Is this just protecting from the possibility of another module using "lockup" as a StoreKey? Why doesn't gravity put everything in its paramSpace keys (some things are set like so `store := ctx.KVStore(k.storeKey)
	store.Set()` )?

Reply:
Parameters are distinct from normal store values. The normal kv store has very few rules, the param store specifically contains parameters that are loaded from the genesis file and registered so that they can be changed by governance.

Think of it as a subset of the generic kv store that requires it's own boilerplate because of the way governance can interact with the data there.